tsc/src/enemies/doom_larry.cpp

```diff
@@ -0,0 +1,168 @@
+/***************************************************************************
+ * doom_larry.cpp - Run or die.
+ *
+ * Copyright © 2014 The TSC Contributors
+ ***************************************************************************/
+/*
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "doom_larry.hpp"
+#include "../core/xml_attributes.hpp"
+#include "../core/game_core.hpp"
+#include "../core/sprite_manager.hpp"
+#include "../core/i18n.hpp"
+#include "../level/level_player.hpp"
+#include "../level/level.hpp"
+#include "../video/animation.hpp"
+#include "../objects/box.hpp"
+
+using namespace TSC;
+
+cDoomLarry::cDoomLarry(cSprite_Manager* p_sprite_manager)
+    : cLarry(p_sprite_manager)
+{
+    Init();
+}
+
+cDoomLarry::cDoomLarry(XmlAttributes& attributes, cSprite_Manager* p_sprite_manager)
+    : cLarry(p_sprite_manager)
+{
+    Init();
+
+    Set_Pos(attributes.fetch<float>("posx", 0), attributes.fetch<float>("posy", 0), true);
+    Set_Direction(Get_Direction_Id(attributes.fetch("direction", Get_Direction_Name(m_start_direction))));
+}
+
+cDoomLarry::~cDoomLarry()
+{
+    //
+}
+
+void cDoomLarry::Init()
+{
+    //The Doom_Larry constructor will have called the Larry constructor, creating an image set full of grey Larry images that we do not want
+    //Remove them so that they can be replaced with Doom Larry images.  If Init() is called again, this clear will also be needed again.
```

Comment:
Please wrap your comments at around 70-80 chars, it gets hard to read otherwise (scrolling right).